rfc/0084-app-service-level-telemetry.md

```diff
@@ -0,0 +1,532 @@
+with an ad-targeting or recommendation targeting deployment, there may be short
+latencies that when not serviced, lead to other processing.  This is worth
+mentioning because it will be important to compare a deployment to itself, and
+there may be limited value in comparing across deployments.
+
+## Capella Service Level Metrics
+
+The counters described above give a good picture of actual impact on a deployed
+app, but they do not give a sense of whether or not a given deployment is
+meeting an expected service level.  Said another way, a Couchbase Capella
+Cluster deployment could be operational, but very slow during a rebalance
+leading to user discontent, but never turn into an error.
+
+Service levels are specific to the user's targets.  While it's impossible to
+properly measure whether or not an end-user's app is meeting service level
+targets, we can establish a predefined histogram that would allow Couchbase to
+simply and inexpensively understand if the service level has likely changed.
+
+It's proposed that we establish a set of standardized histograms, specific to
+services and to the service level expected with the service using those
+parameters.  Counters for these histograms can be sent to Couchbase and both
+stored individually and aggregated.
+
+As an aggregated set of metrics, it would give Couchbase a rough view over the
+service level the user is experiencing.  Deviations in service level can be dug
+into and, in particular, we can check for correlation to Couchbase Capella
+management activities like upgrades.
+
+This list represents the services, buckets.
+
+* Data Service, Non-Durable
+  * Retrieval
+    * <1ms
+    * 1-10ms
+    * 10-100ms
+    * 100-500ms
+    * 500-1000ms
+    * 1s-2.5s
+    * \>2.5s
+  * Mutation
+    * <1ms
+    * 1-10ms
+    * 10-100ms
+    * 100-500ms
+    * 500-1000ms
+    * 1s-2.5s
+    * \>2.5s
+* Data Service, Durable
+  * Mutation
+    * 1-10ms
+    * 10-100ms
+    * 100-1000ms
+    * 1-2s
+    * 2s-5s
+    * 5s-10s
+    * \>10s
+* Query Service
+  * Query (all)
+    * 0-100ms
+    * 100-1000ms
+    * 1s-10s
+    * 10s-30s
+    * 30-75s
+    * \>75s
+* FTS Service
+  * Search (all)
+    * 0-100ms
+    * 100-1000ms
+    * 1s-10s
+    * 10s-30s
+    * 30-75s
+    * \>75s
+* Analytics Service
+  * Query (all)
+    * 0-100ms
+    * 100-1000ms
+    * 1s-10s
+    * 10s-30s
+    * 30-75s
+    * \>75s
+
+The Indexing Service and Eventing Services are not included as they are not
+network-available services for applications to make requests from.
+
+As with the Service Experience Counters above, it may be incorrect to try to
+compare between deployments.  For example, the query complexity may differ
+significantly or the data size and use of features like compression may vary
+significantly between two applications.
+
+# API and Implementation Details
+
+The decision to reuse the existing Meter implementation is considered an SDK
```

Comment:
N.b. this is could be tricky as the Meter interface as specced is very limited, only supporting histograms (no counters).
Perhaps we should consider allowing SDKs to track the metrics internally using whatever implementation they choose.

Reply:
This is what this RFC is suggesting while saying 
> The application service-level meter would continue to function even if the user provides a custom meter implementation.

Reply:
Ok.  Personally I would remove mentions of reusing Meter implementation as I don't think that's possible and so it's just adding more confusion than signal, but up to you - will resolve.